nixos/modules/image/repart-verity-store.nix

```diff
@@ -0,0 +1,198 @@
+# opinionated module that can be used to build nixos images with
+# a dm-verity protected nix store and a volatile rootfs
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  cfg = config.image.repart.verityStore;
+
+  verityMatchKey = "store";
+
+  # TODO: make these and other arch mappings available from systemd-lib for example
+  partitionTypes = {
+    usr =
+      {
+        "x86_64" = "usr-x86-64";
+        "arm64" = "usr-arm64";
+      }
+      ."${pkgs.stdenv.hostPlatform.linuxArch}";
+
+    usr-verity =
+      {
+        "x86_64" = "usr-x86-64-verity";
+        "arm64" = "usr-arm64-verity";
+      }
+      ."${pkgs.stdenv.hostPlatform.linuxArch}";
+  };
+
+  ukiPath = "/EFI/Linux/${config.system.boot.loader.ukiFile}";
+
+  verityHashCheck =
+    pkgs.buildPackages.writers.writePython3Bin "assert_uki_repart_match.py"
+      {
+        flakeIgnore = [ "E501" ]; # ignores PEP8's line length limit of 79 (black defaults to 88 characters)
+      }
+      (
+        builtins.replaceStrings [ "@NIX_STORE_VERITY@" ] [
+          partitionTypes.usr-verity
+        ] (builtins.readFile ./assert_uki_repart_match.py)
```

Comment:
```suggestion
        builtins.replaceVars ./assert_uki_repart_match.py {
          NIX_STORE_VERITY = partitionTypes.usr-verity;
        }
```
nit

Reply:
huh, this isn't a builtin, is it?

Reply:
Right. I made this suggestion from the GH editor. This is [`pkgs.replaceVars`](https://noogle.dev/f/pkgs/replaceVars), actually

Reply:
Personally, I'd prefer to keep the approach using builtins, given that this replacement is simple enough and the source file is in-tree